KH-107: Assign encounter privilege appriopriatley to HC & Hospital roles

kdaud opened the pull request:
Refer to [KH-107](https://issues.openmrs.org/browse/KH-107)

Ruhanga commented on configs/openmrs_config/roles/cambodia_roles.csv:
```diff
@@ -1,2 +0,0 @@
-Uuid,Role name,Description,Inherited roles,Privileges,_order:1100
```
I'd leave the file name untouched to avoid potential collusion from parent configs.

Ruhanga commented on configs/openmrs_config/roles/cambodia_roles.csv:
```diff
@@ -1,2 +1,4 @@
-a28fc3ff-cf23-44e6-9b4b-a13aee22363b,NCD User,NCD User,,Manage Concept Classes; Delete Patient Identifiers; Configure Visits; Preview Forms; Manage Privileges; Get Order Sets; View Navigation Menu; Delete Cohorts; Manage Forms; Delete Patient Programs; Manage Field Types; Add Observations; View Encounters; View Problems; Delete HL7 Inbound Exception; Update HL7 Source; View Metadata Via Mapping;  ...
\ No newline at end of file
+a28fc3ff-cf23-44e6-9b4b-a13aee22363b,NCD care provider,NCD care provider,,Manage Concept Classes; Delete Patient Identifiers; Configure Visits; Preview Forms; Manage Privileges; Get Order Sets; View Navigation Menu; Delete Cohorts; Manage Forms; Delete Patient Programs; Manage Field Types; Add Observations; View Encounters; View Problems; Delete HL7 Inbound Exception; Update HL7 Source; View Meta ...
+8c26a98b-fa03-4c37-b788-6d1e537d4071,NCD care provider in Health Center,NCD care provider in Health Center,,View NCD Screening encounters; Edit NCD Screening encounters;View NCD Medical History encounters;View NCD Consultation encounters,
+c65ad447-bbe8-4917-abc6-30617dbac994,NCD care provider in Hospital,NCD care provider in Hospital,,View NCD Medical History encounters;Edit NCD Medical History encounters;View NCD Consultation encounters;Edit NCD Consultation encounters;Vew NCD Screening encounters,
```
@kdaud, according to the ticket description, you should use the `Inherited roles` field to inherit from the required roles instead of adding their corresponding individual privileges.

Ruhanga commented on configs/openmrs_config/roles/cambodia_roles.csv:
```diff
@@ -1,2 +1,4 @@
-a28fc3ff-cf23-44e6-9b4b-a13aee22363b,NCD User,NCD User,,Manage Concept Classes; Delete Patient Identifiers; Configure Visits; Preview Forms; Manage Privileges; Get Order Sets; View Navigation Menu; Delete Cohorts; Manage Forms; Delete Patient Programs; Manage Field Types; Add Observations; View Encounters; View Problems; Delete HL7 Inbound Exception; Update HL7 Source; View Metadata Via Mapping;  ...
\ No newline at end of file
+a28fc3ff-cf23-44e6-9b4b-a13aee22363b,NCD care provider,NCD care provider,,Manage Concept Classes; Delete Patient Identifiers; Configure Visits; Preview Forms; Manage Privileges; Get Order Sets; View Navigation Menu; Delete Cohorts; Manage Forms; Delete Patient Programs; Manage Field Types; Add Observations; View Encounters; View Problems; Delete HL7 Inbound Exception; Update HL7 Source; View Meta ...
+8c26a98b-fa03-4c37-b788-6d1e537d4071,NCD care provider in Health Center,NCD care provider in Health Center,NCD care provider,View NCD Screening encounters;Edit NCD Screening encounters;View NCD Medical History encounters;View NCD Consultation encounters,
+ffba7d63-c1cf-470f-8a58-ad393c42fc98,NCD care provider in Hospital,NCD care provider in Hospital,NCD care provider,View NCD Medical History encounters;Edit NCD Medical History encounters;View NCD Consultation encounters;Edit NCD Consultation encounters;Vew NCD Screening encounters,
```
@kdaud, do you need to add the `View` and `Edit` privileges if those are part of the associated inherited roles?

kdaud replied:
@Ruhanga have omitted  them.

Ruhanga replied:
You didn't have to remove them. Sorry I didn't mean to mislead you this way. https://github.com/openmrs/ozone-distro-cambodia/pull/103/commits/9594cfb409e7291e1895bb48020c5ea611f18623 is not necessary. I just wanted you to be sure you know what you are doing. :-)

kdaud replied:
> I just wanted you to be sure you know what you are doing. :-)

I knew these specific privileges were required for the respective added roles. However, your [comment](https://github.com/openmrs/ozone-distro-cambodia/pull/103#discussion_r1181636561) made me cast a vote of dought.

rbuisson replied:
@kdaud , this privilege is not created. See the typo here: `Vew NCD Screening encounters`

rbuisson commented on configs/openmrs_config/roles/cambodia_roles.csv:
```diff
@@ -1,2 +1,4 @@
 Uuid,Role name,Description,Inherited roles,Privileges,_order:1100
-a28fc3ff-cf23-44e6-9b4b-a13aee22363b,NCD User,NCD User,,Manage Concept Classes; Delete Patient Identifiers; Configure Visits; Preview Forms; Manage Privileges; Get Order Sets; View Navigation Menu; Delete Cohorts; Manage Forms; Delete Patient Programs; Manage Field Types; Add Observations; View Encounters; View Problems; Delete HL7 Inbound Exception; Update HL7 Source; View Metadata Via Mapping;  ...
\ No newline at end of file
+a28fc3ff-cf23-44e6-9b4b-a13aee22363b,NCD care provider,NCD care provider,,Manage Concept Classes; Delete Patient Identifiers; Configure Visits; Preview Forms; Manage Privileges; Get Order Sets; View Navigation Menu; Delete Cohorts; Manage Forms; Delete Patient Programs; Manage Field Types; Add Observations; View Encounters; View Problems; Delete HL7 Inbound Exception; Update HL7 Source; View Meta ...
```
Could we make this title case? (Sorry I know this is not title case in the ticket description, but I feel it will look better).

rbuisson reviewed (changes requested):
Left a typo in the roles file: `Vew NCD Screening encounters`
Therefore one role is not created.

![Snip20230502_1](https://user-images.githubusercontent.com/10692811/235748718-6446d848-64a5-430e-9847-efe82fa17e87.png)

Have you tested this @kdaud ? First thing I checked and found that one privilege is not created.

kdaud commented:
@rbuisson, have addressed the reviews and hopefully we are good to go.

![Roles](https://user-images.githubusercontent.com/38831673/235762631-c3ac2dc2-888a-4b95-99ae-4ccb9df0bbb9.png)
